crates/gitbutler-project-store/src/lib.rs

```diff
@@ -0,0 +1,92 @@
+use anyhow::{anyhow, Context, Result};
+use migrations::{gitbutler_migrations::gitbutler_migrations, migrator::Migrator};
+use rusqlite::Connection;
+use std::path::Path;
+
+mod changes;
+mod migrations;
+
+const DATABASE_NAME: &str = "project.sqlite";
+
+/// ProjectStore provides a light wrapper around a sqlite database
+struct ProjectStore {
+    connection: Connection,
+}
+
+impl ProjectStore {}
+
+/// Database setup
+///
+/// Before touching any database related code, please read https://github.com/the-lean-crate/criner/discussions/5 first.
+impl ProjectStore {
+    /// Creates an instance of ProjectStore and runs any pending sqlite migrations
+    /// gitbutler_project_directory should be the `.git/gitbutler` path of a given
+    /// repository
+    pub fn initialize(gitbutler_project_directory: &Path) -> Result<ProjectStore> {
+        let database_path = gitbutler_project_directory.join(DATABASE_NAME);
+        let database_path = database_path.to_str().ok_or(anyhow!(
+            "Failed to get database {}",
+            gitbutler_project_directory.display()
+        ))?;
+
+        let connection = Connection::open(database_path)?;
+
```

Comment:
This whitespace puzzles me, even though I know that I had a phase where I did that too.